torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -108,6 +213,11 @@ def addmm_pattern(
 
 
 def should_pad_addmm(match: Match) -> bool:
+    if (
+        torch._inductor.config.keep_output_stride
+        and _result_layout_affects_graph_output(match)
+    ):
+        return False
```

Comment:
Maybe I'm missing something but can we just check if the output of the addmm is a graph output ?

Reply:
If we would be on the IR level, I think that would be sufficient if we unpack views. But since we're looking at the FX graph, I imagine the GEMM memory output layout could propagate through node fusions. But I'm not sure if this is an entirely realistic concern, since a GEMM followed by a slice and then followed by a Pointwise op would not be fused with the Pointwise because the slicing means that the Pointwise op should not be applied to the entirety of the GEMM output. So, yes, maybe we can simplify this a lot, but I'm unsure if it's safe at the moment, and pretty sure it's hard to guarantee it stays safe.

Reply:
Non-dense format will only propagate to output through views. Check examples here of looking at uses/views https://github.com/pytorch/pytorch/blob/ea7149aa2233b7987aadc3d6e2ba94cd3e133606/torch/_inductor/utils.py#L306-L326.

Reply:
Thanks for the code pointer, that allowed me to simplify it a lot..